### DISCUSSION TITLE
How to detect irregular table

### DISCUSSION THREAD

**Discussion by Godlikemandyy** [jsvine/pdfplumber#617] [Ask for help with specific PDFs]
I use .debug_tablefinder() to detect table, I find the result is incomplete, eg:
![image](https://user-images.githubusercontent.com/36065226/155111495-afc1f98d-2e9d-4bd3-83c3-1880bf3f8b1b.png)
How can I get complete result?

**Comment by jsvine** [comment]
Hi @Godlikemandyy, and thanks for your interest in this library. I've converted your inquiry into a "discussion," since you're looking for help with a specific PDF, rather than proposing a general solution or identifying a bug. 

In this case, the table you're trying to extract is missing it's borders on the left and right sides of the content. 

The most reliable way to capture the full content is to pass `"explicit_vertical_lines": [ a, b ],` to your table settings, where `a` and `b` are the x-coordinates of the left and right side of the table. You can determine those coordinates *manually* or you can calculate those values by examining the `x0` and `x1` properties of the horizontal lines in the table.

**Reply by Godlikemandyy** [reply]
@jsvine Thanks for your patient reply！I have tried this method to solve it，it is effective. But I have another problem, a pdf file has many pages, each page has some tables and these tables are in different places on the page so `a` and `b` must be different. I want to automatically find them, what should I do? Looking forward to your reply, thanks!

**Reply by jsvine** [reply]
The precise approach will depend on the specific PDFs you're working with — how they're formatted, how consistent they are, et cetera. If the PDF uses the same table formatting throughout, for example, your best bet is probably to search through `page.lines` to find the wide horizontal lines, and to use their positioning to determine the `a` and `b` parameters.

**Reply by Godlikemandyy** [reply]
The pdf file has diffenrent table format，so it's too hard to search the same `a`  and `b`!
